handlers/rootfs_image.go

```diff
@@ -139,7 +140,26 @@ func (rp *Rootfs) ReadHeader(r io.Reader, path string, version int, augmented bo
 		}
 
 	case filepath.Base(path) == "meta-data":
-		// TODO: implement when needed
+		dec := json.NewDecoder(r)
+		var data interface{}
+		err := dec.Decode(&data)
+		if err == io.EOF {
```

Comment:
But then we would have to `Read` the data first, which would again call for this check.

FWIW, code coverage shows that the tests do not enter this block, which might indicate that it's not necessary. I suspect it might trigger on an empty `meta-data` file though. I can experiment a bit.